SmartDeviceLink/SDLSystemCapabilityManager.h

```diff
@@ -239,53 +248,88 @@ typedef void (^SDLCapabilityUpdateHandler)(SDLSystemCapability *capability);
+
+/**
+ * Subscribe to a particular capability type with a selector callback.
+ *
+ * The selector supports the following parameters:
  *
  * 1. No parameters e.g. `- (void)phoneCapabilityUpdated;`
- * 2. One `SDLSystemCapability *` parameter e.g. `- (void)phoneCapabilityUpdated:(SDLSystemCapability *)capability`
  *
- * This method will be called immediately with the current value and called every time the value is updated on RPC v5.1.0+ systems (`supportsSubscriptions == YES`). If this method is called on a sub-v5.1.0 system (`supportsSubscriptions == NO`), the method will return `NO` and the selector will never be called.
+ * 2. One `SDLSystemCapability *` parameter, e.g. `- (void)phoneCapabilityUpdated:(SDLSystemCapability *)capability`
+ *
+ * 3. Two parameters, one `SDLSystemCapability *` parameter, and one `NSError` parameter, e.g. `- (void)phoneCapabilityUpdated:(SDLSystemCapability *)capability error:(NSError *)error`
+ *
+ * 4. Three parameters, one `SDLSystemCapability *` parameter, one `NSError` parameter, and one `BOOL` parameter e.g. `- (void)phoneCapabilityUpdated:(SDLSystemCapability *)capability error:(NSError *)error subscribed:(BOOL)subscribed`
+ *
+ * On v5.1.0+ systems (where `supportsSubscriptions == YES`):
+ * This method will be called immediately with the current value and will be called every time the value is updated. If this is the first subscription of this `SDLSystemCapabilityType`, then the value will be retrieved and a subscription will be attempted. The current cached value (`nil`) will nevertheless be returned immediately.
+ *
+ * On sub-v5.1.0 systems (where `supportsSubscriptions == NO`):
+ * The method will be called immediately with the current value and will _not_ be automatically called every time the value is updated, unless the `type` is `DISPLAYS` which is supported on every version. If `updateCapabilityType:completionHandler` is called and a new value is retrieved, this value will be updated then. If this is the first subscription of this `SDLSystemCapabilityType`, then the value will be retrieved and returned. The current cached value (`nil`) will nevertheless be returned immediately.
  *
  * @param type The type of the system capability to subscribe to
  * @param observer The object that will have `selector` called whenever the capability is updated
  * @param selector The selector on `observer` that will be called whenever the capability is updated
- * @return Whether or not the subscription succeeded. `NO` if the connected system doesn't support capability subscriptions, or if the `selector` doesn't support the correct parameters (see above)
+ * @return Whether or not the subscription succeeded. `NO` if the connected system doesn't support capability subscriptions, or if the `selector` doesn't support the correct parameters (see above).
  */
 - (BOOL)subscribeToCapabilityType:(SDLSystemCapabilityType)type withObserver:(id)observer selector:(SEL)selector;
```

Comment:
If I attempt to subscribe to `SDLSystemCapabilityTypeVideoStreaming` after I have got the notification that the SDL app started successfully but before the `hmiLevel` is `FULL` then I get an error saying `SDLResultDisallowed` but if I wait until the `hmiLevel` is `FULL` then I get the video capabilities from the module (`appType` is `default`). Maybe add some documentation that subscriptions should only happened if the `hmiLevel` is non-`NONE`.

Reply:
Fixed, now it will return `nil` or `NO` if it fails to attempt a subscription and tracks the HMI level and cancels out if the HMI level is NONE.

Reply:
Fixed, the HMI level now starts assuming NONE and will change when it gets an explicit HMI status.